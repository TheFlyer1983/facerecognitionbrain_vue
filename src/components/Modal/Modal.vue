<script setup lang="ts">
const emits = defineEmits(['close-modal']);

function closeModal() {
  emits('close-modal');
}
</script>

<template>
  <div class="modal position-abosolute d-flex justify-content-center align-items-center">
    <div
      class="modal__overlay position-fixed w-100 h-100 bg-white-80 top-0 start-0"
      @click="closeModal"
      data-test="close-modal"
    ></div>
    <div class="modal__container">
      <slot></slot>
    </div>
  </div>
</template>

<style lang="scss" scoped>
.modal {
  &__overlay {
    z-index: 100;
  }

  &__container {
    z-index: 110;
  }
}
</style>
