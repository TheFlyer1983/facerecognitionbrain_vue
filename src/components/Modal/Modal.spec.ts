import { shallowMount, VueWrapper } from '@vue/test-utils';
import { describe, beforeEach, it, expect, vi, beforeAll } from 'vitest';

import Modal from './Modal.vue';

describe('Given the modal component', () => {
  const render = () => shallowMount(Modal);
  let wrapper: VueWrapper<InstanceType<typeof Modal>>;

  describe('when it is rendered', () => {
    beforeEach(() => {
      wrapper = render();
    });

    it('should match the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    describe('and when the overlay is clicked', () => {
      beforeEach(() => {
        wrapper.find('[data-test="close-modal"]').trigger('click');
      });

      it('should emit the correct event', () => {
        expect(wrapper.emitted('close-modal')).toStrictEqual([[]])
      })
    });
  });
});
