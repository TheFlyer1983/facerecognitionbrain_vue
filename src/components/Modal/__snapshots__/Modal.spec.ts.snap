// Vitest Snapshot v1

exports[`Given the modal component > when it is rendered > should match the snapshot 1`] = `
<div
  class="modal position-abosolute d-flex justify-content-center align-items-center"
  data-v-907877e9=""
>
  <div
    class="modal__overlay position-fixed w-100 h-100 bg-white-80 top-0 start-0"
    data-test="close-modal"
    data-v-907877e9=""
  />
  <div
    class="modal__container"
    data-v-907877e9=""
  >
    
    
  </div>
</div>
`;
